src/tools/Avalonia.Generators/Common/XamlXNameResolver.cs

```diff
@@ -89,4 +90,9 @@ private string TryGetFieldModifier(XamlAstObjectNode objectNode)
             _ => _defaultFieldModifier
         };
     }
+
+    private static bool IsAttachedProperty(XamlAstNamePropertyReference namedProperty)
+    {
+        return !namedProperty.DeclaringType.Equals(namedProperty.TargetType);
```

Comment:
Shouldn't it be IsAssignableFrom ?

Reply:
If I use this:
```csharp
private static bool IsAttachedProperty(XamlAstNamePropertyReference namedProperty)
{
    return !namedProperty.DeclaringType.GetClrType().IsAssignableFrom(namedProperty.TargetType.GetClrType());
}
```
then the test `XamlXNameResolverTests.Should_Resolve_Types_From_Avalonia_Markup_File_When_Types_Contains_Generic_Arguments` fails.

Everything works fine with the current approach. Here it does not seem to matter that `Name` comes from `StyledElement`, otherwise the tests would fail. I also tested it with the actual nuget packages from the PRBUILD, it solves the problem but still generates the normal field for direct `Name` property.

It would maybe make sense to have a `IsAttachedProperty` method or extension method directly on `XamlAstNamePropertyReference`, but I guess that would need to go into `XamlX`.